File: java/src/main/java/com/meteroid/models/CreditNoteStatus.java
// This file is @generated
package com.meteroid.models;

import com.fasterxml.jackson.annotation.JsonValue;
import com.meteroid.Utils.ToQueryParam;

public enum CreditNoteStatus implements ToQueryParam {
    DRAFT("DRAFT"),
    FINALIZED("FINALIZED"),
    VOIDED("VOIDED");
    private final String value;

    CreditNoteStatus(String value) {
        this.value = value;
    }

    @JsonValue
    public String getValue() {
        return this.value;
    }

    @Override
    public String toQueryParam() {
        return this.value;
    }
}

File: go/credit_note_status.go
// this file is @generated
package meteroid

type CreditNoteStatus string

// Known values of CreditNoteStatus.
//
// The API may add new values over time; unknown values round-trip unchanged.
const (
	CreditNoteStatusDraft     CreditNoteStatus = "DRAFT"
	CreditNoteStatusFinalized CreditNoteStatus = "FINALIZED"
	CreditNoteStatusVoided    CreditNoteStatus = "VOIDED"
)

// AllCreditNoteStatusValues lists every CreditNoteStatus value known to this SDK version.
var AllCreditNoteStatusValues = []CreditNoteStatus{
	CreditNoteStatusDraft,
	CreditNoteStatusFinalized,
	CreditNoteStatusVoided,
}

// String returns the wire representation of the value.
func (e CreditNoteStatus) String() string {
	return string(e)
}

// IsKnown reports whether the value is one this SDK version knows about.
func (e CreditNoteStatus) IsKnown() bool {
	for _, known := range AllCreditNoteStatusValues {
		if e == known {
			return true
		}
	}
	return false
}
